### ISSUE TITLE
Allow throwing away all changes instead of uploading them

### ISSUE THREAD

**Issue by jidanni** [openstreetmap/iD#10033]
### Description

Vespucci allows throwing away all changes instead of uploading them. iD should too. Yes, after a "are you sure?" screen.

You might say
> Just undo undo undo. Voila, no more changes to upload.

But as we see in #10029, there might be no full undo list.
So the user is stuck with changes they can't get rid of unless they upload them, or make a new OSM account.

> Wait until #10031 is implemented.

But that is hard to implement, whereas throwing away all changes at once is easy to implement.

So at the Save > Upload / Cancel screen, a third choice "Throw away all these edits" etc. should appear.

**Comment by k-yle** [comment]
as a workaround I usually just press the browser's refresh button

**Comment by jidanni** [comment]
Yes, using the unofficial Refresh the page method will give the user an X to wipe their changes off, until one day as in #10029 it doesn't work... Therefore there should be an official way to shake off one's edits instead of uploading them.

**Comment by Asif-Sheriff** [comment]
Heyy 
I have been working on a functionality like you suggested, eventually decided on a discard button next to the save button.

You can see it in action here:
[Screencast from 12-12-23 10:52:08 PM IST.webm](https://github.com/openstreetmap/iD/assets/130328580/8f4f37d4-d138-456d-bb38-63dd37730568)

Features:
If a user wants to discard all the changes without saving them, they may do so without having to reload the page.
The user if they want can also undo the discard action(which is not possible by reloading).

Also if you have any suggestions feel free to tell me.

Regards

**Comment by jidanni** [comment]
Thanks @Asif-Sheriff !
However:
1. I think discard is too dangerous an action to have sitting there. Like having a gun in one's house. I think it should only appear when the final save dialog is running...
Wait. You're right. He doesn't want to push Save in the first place. OK, make sure it has a lot of "are you sure?" double confirmations.
2. Also I don't think this should get tangled up with the undo system... which might have caused the mess the user it trying to get out of. But I'm not sure.

Anyway thanks for your hard work, and I hope an administrator will approve your changes.

**Comment by Asif-Sheriff** [comment]
The discard button here does not have a confirmation screen so I thought the user might want to undo changes if they discarded them accidentally.

In any case thanks for your input I will wait for admins to take a look at this before I make a PR.

**Comment by tyrasd** [comment]
closing as low-impact, see https://github.com/openstreetmap/iD/pull/10045#issuecomment-3743495349
